ui/helpers/utils/accounts.js

```diff
@@ -73,43 +74,81 @@ export function getAvatarNetworkColor(name) {
   }
 }
 
-export function getAccountLabel(
+const toSrpLabel = (index) =>
+  // Index starts at 1, for SRPs.
+  `SRP #${index + 1}`;
+
+export function getAccountLabels(
   type,
   account,
+  keyrings,
   ///: BEGIN:ONLY_INCLUDE_IF(keyring-snaps)
   snapName,
-  snapPreinstalled,
   ///: END:ONLY_INCLUDE_IF
 ) {
   if (!account) {
-    return null;
+    return [];
   }
+
+  const labels = [];
+
+  const hdKeyrings = keyrings.filter(
+    (keyring) => keyring.type === KeyringType.hdKeyTree,
+  );
+
   switch (type) {
-    case KeyringType.hdKeyTree:
-      return null;
+    case KeyringType.hdKeyTree: {
+      if (hdKeyrings.length > 1) {
+        const hdKeyringIndex = hdKeyrings.findIndex((kr) =>
+          kr.accounts.includes(account.address),
+        );
+        labels.push(toSrpLabel(hdKeyringIndex));
+      }
+      break;
+    }
     case KeyringType.imported:
-      return t('imported');
+      labels.push(t('imported'));
+      break;
     case KeyringType.qr:
-      return HardwareKeyringNames.qr;
+      labels.push(HardwareKeyringNames.qr);
+      break;
     case KeyringType.trezor:
-      return HardwareKeyringNames.trezor;
-    case KeyringType.oneKey:
-      return HardwareKeyringNames.oneKey;
+      labels.push(HardwareKeyringNames.trezor);
+      break;
     case KeyringType.ledger:
-      return HardwareKeyringNames.ledger;
+      labels.push(HardwareKeyringNames.ledger);
+      break;
+    case KeyringType.oneKey:
+      labels.push(HardwareKeyringNames.oneKey);
+      break;
     case KeyringType.lattice:
-      return HardwareKeyringNames.lattice;
+      labels.push(HardwareKeyringNames.lattice);
+      break;
     ///: BEGIN:ONLY_INCLUDE_IF(keyring-snaps)
-    case KeyringType.snap:
-      if (snapPreinstalled) {
-        return null;
+    case KeyringType.snap: {
+      const { entropySource } = account.options;
+      if (entropySource) {
+        const hdKeyringIndex = hdKeyrings.findIndex(
+          (kr) => kr.metadata.id === entropySource,
+        );
+        labels.push(toSrpLabel(hdKeyringIndex));
+      }
```

Comment:
Question: Should we also check for `hdKeyrings.length > 1` here too? Cause I think that if we create a Snap account that has a `options.entropySource`, then we would add a pill with `SRP #1` too.

Is that ok?